auth/auth_test.go

```diff
@@ -0,0 +1,260 @@
+		{ConfigPath: cp[0], id: "abc-123", Pass: false},
+
+		{ConfigPath: cp[1], id: "agent-123", Pass: true},
+		{ConfigPath: cp[1], id: "judge-123", Pass: true},
+		{ConfigPath: cp[1], id: "logger-123", Pass: true},
+
+		{ConfigPath: cp[2], id: "agent-123", Pass: true},
+		{ConfigPath: cp[2], id: "logger-123", Pass: true},
+		{ConfigPath: cp[2], id: "judge-123", Pass: true},
+
+		{ConfigPath: cp[2], id: "abc-123", Pass: false},
+		{ConfigPath: cp[6], id: "judge-123", Pass: false},
+		{ConfigPath: cp[6], id: "logger-123", Pass: false},
+		{ConfigPath: cp[7], id: "", Pass: false},
+	}
+
+	for tc, _ := range testCases {
+		config, err1 := config.NewConfig(testCases[tc].ConfigPath)
+		if err1 != nil {
+			PrintResult(t, testCases[tc].Pass, err1)
+		} else {
+			au, err2 := NewAuth(config)
+			if err2 != nil {
+				PrintResult(t, testCases[tc].Pass, err2)
+			} else {
+				err3 := au.Register(testCases[tc].id)
+				if err3 != nil {
+					PrintResult(t, testCases[tc].Pass, err3)
+				} else {
+					PrintResult(t, testCases[tc].Pass, err2)
+				}
+			}
+		}
+	}
+}
+
+func TestAuth(t *testing.T) {
+	testCases := []TestInput{
+		{ConfigPath: cp[0], id: "", Pass: false},
+		{ConfigPath: cp[0], id: "agent-123", Pass: true},
+		{ConfigPath: cp[0], id: "judge-123", Pass: false},
+		{ConfigPath: cp[0], id: "logger-123", Pass: true},
+		{ConfigPath: cp[0], id: "abc-123", Pass: false},
+
+		{ConfigPath: cp[1], id: "agent-123", Pass: false},
+		{ConfigPath: cp[1], id: "judge-123", Pass: true},
+		{ConfigPath: cp[1], id: "logger-123", Pass: true},
+
+		{ConfigPath: cp[2], id: "agent-123", Pass: false},
+		{ConfigPath: cp[2], id: "logger-123", Pass: true},
+		{ConfigPath: cp[2], id: "judge-123", Pass: true},
+		{ConfigPath: cp[2], id: "abc-123", Pass: false},
+
+		{ConfigPath: cp[3], id: "agent-123", Pass: false},
+		{ConfigPath: cp[3], id: "logger-123", Pass: false},
+		{ConfigPath: cp[3], id: "judge-123", Pass: false},
+
+		{ConfigPath: cp[6], id: "judge-123", Pass: false},
+		{ConfigPath: cp[6], id: "logger-123", Pass: false},
+		{ConfigPath: cp[7], id: "", Pass: false}, //idk
+	}
+	for tc, _ := range testCases {
+		config, err1 := config.NewConfig(testCases[tc].ConfigPath)
+		if err1 != nil {
+			PrintResult(t, testCases[tc].Pass, err1)
+		} else {
+			au, err2 := NewAuth(config)
+			if err2 != nil {
+				PrintResult(t, testCases[tc].Pass, err2)
+			} else {
+				err3 := au.Auth(testCases[tc].id, methodCases[tc], &objectCases[tc])
+				if err3 != nil {
+					PrintResult(t, testCases[tc].Pass, err3)
+				} else {
+					PrintResult(t, testCases[tc].Pass, err2)
+				}
+			}
+		}
+	}
+}
+
+type TestInput1 struct {
+	tip    TestInput
+	id     string
+	method object.Method
+	key    object.Key
+	Pass   bool
+}
+
+func TestHash(t *testing.T) {
+	testCases := []TestInput1{
+		{tip: TestInput{ConfigPath: cp[0]}, id: "agent-123", method: object.MethodGet, key: object.Key{Type: "world", Name: "judge", Namespace: "default"}, Pass: true},
```

Comment:
It's better to use the variables you declared at the end of this file - methodCases